version: "3.9"

networks:
  pg-network:
    external: true

services:
  pgdatabase:
    container_name: pgsql-dev
    image: postgres:13
    healthcheck:
      test: [ "CMD", "pg_isready" ]
      interval: 10s
      timeout: 45s
      retries: 10
      start_period: 30s
    volumes:
      - /Users/waleed/development/docker/dataeng/data:/var/lib/postgresql/data:rw
    environment:
      - POSTGRES_DB=nytaxi
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=root
    ports:
      - "5433:5432"
    restart: always
    networks:
      - pg-network

  pgadmin:
    container_name: pgadmin-dev
    image: dpage/pgadmin4:latest
    volumes:
      - /Users/waleed/development/docker/dataeng/pgadmin:/var/lib/pgadmin
    environment:
      - PGADMIN_DEFAULT_EMAIL=admin@admin.com
      - PGADMIN_DEFAULT_PASSWORD=root
      - PGADMIN_LISTEN_ADDRESS=0.0.0.0
    ports:
      - 8080:80
    restart: always
    networks:
      - pg-network
    depends_on:
      - pgdatabase
